addons/mail/models/mail_thread.py

```diff
@@ -4154,24 +4154,34 @@ def _get_mail_redirect_suggested_company(self):
             return self.company_id
         return False
 
+    def _get_mail_thread_access_rights(self):
+        res = {}
+        for thread in self:
+            access_rights = {
+                'hasWriteAccess': False,
+                'hasReadAccess': True,
+                'canPostOnReadonly': thread._mail_post_access == 'read',
+            }
+            try:
+                thread.check_access_rule("write")
+                thread.check_access_rights("write")
```

Comment:
Actually, would be better to use ``filter_access_rights(_python)`` to do it in batch, instead of checking access one record at a time.